main/observing/observation_run.py

```diff
@@ -114,6 +123,13 @@ def everything_ok(self):
         if message:
             logging.error('Hardware connection timeout: {}'.format(message))
 
+        self.monitor.monitor_run(self.th_dict)
+
+        print('---Checking Threads--- LINE 128')
+        self.threadcheck()
+
+        self.threadcheck()
```

Comment:
I'm not sure if just scattering these about everywhere in observation_run is the best approach here, though I can't immediately think of a better alternative either since the threads do need to be restarted from observation_run itself.  In any case, I don't think you need 2 in a row here.  Maybe just put one after every weather check or something.